packages/pwa-kit-extension-sdk/src/shared/utils/extensibility.ts

```diff
@@ -165,3 +169,28 @@ const getPreviousExtensions = (
     const index = array.findIndex((extension) => extension[0] === currentExtension[0])
     return array.slice(index + 1)
 }
+
+/**
+ * For the given extension, merge its user-defined config and default config
+ */
+export const mergeConfigs = (
+    extension: ApplicationExtensionEntryTuple,
+    defaultConfig = getExtensionDefaultConfig(extension[0])
```

Comment:
I'm not quite understanding what is going on here. I've never seen this pattern of using functions argument in the default for another.

Reply:
Lets consider refactoring this for better clarity.